src/runtime/cache.cpp

```diff
@@ -92,15 +92,34 @@ WEAK bool bounds_equal(const buffer_t &buf1, const buffer_t &buf2) {
 }
 
 // Each host block has extra space to store extra information just
-// before the contents.  16 is chosen to keep that alignment. For a
-// buffer between the first lookup and store, this holds the cache key
-// hash. For buffers where the lookup succeeded or the store has
-// occurred, this holds a pointer to the hash entry.
+// before the contents.  16 is chosen to keep that alignment.
+// For buffers where the lookup succeeded or the store has occurred, the first
+// 8 bytes holds a pointer to the hash entry. The second 8 bytes holds the cache
+// key hash.
 //
 // This is an optimization the number of cycles it takes for the cache
 // to operate.
 const size_t extra_bytes_host_bytes = 16;
 
+enum BlockHeaderType {
+    BLOCK_HEADER_ENTRY = 0,
+    BLOCK_HEADER_HASH = 1
+};
+
+WEAK uint8_t *get_pointer_to_header(uint8_t * host, BlockHeaderType type) {
```

Comment:
I think this would be better if the header were a two-field struct containing the entry pointer and the hash key, and then get_pointer_to_header just returned a pointer to the struct instead of taking an enum as the second arg.